99-other-topics/008-zero-downtime-deployment.md

```diff
@@ -29,5 +29,27 @@ The ability to constantly update your application, with zero downtime, brings a
 
 If you'd like to read more in-depth into rolling updates, then Kubernetes has some great documentation [here](https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/).
 
+### Readiness/Liveness probes and SSL in Rails applications
 
+For zero downtime deployments, you will need a readiness probe in your application, so that the cluster knows when your container is ready to receive traffic. You will also need a liveness probe, so that the cluster knows if it needs to restart your container after a crash.
+
+SSL will be terminated outside of your pod, so your probes will need to respond to HTTP requests. However, Ruby on Rails applications are sometimes configured to only respond to HTTPS traffic (by adding `config.force_ssl = true` in the `config/environments/production.rb` file).
+
+In this case, the application will respond to any HTTP request with a 301 redirect, asking the requester to resend the request to the equivalent HTTPS URL. This will cause your probes to fail, because the redirect will not be followed.
```

Comment:
Status code varies by Rails version. In 5.0+ it does the Right Thing™ and sometimes sends a 307. Maybe omit?

Reply:
Good point. I'll update.